src/ebmc/transition_system.cpp

```diff
@@ -368,3 +368,53 @@ int show_symbol_table(
   return get_transition_system(
     cmdline, message_handler, dummy_transition_system);
 }
+
+std::vector<symbol_exprt> transition_systemt::state_variables() const
+{
+  std::vector<symbol_exprt> state_variables;
+
+  const auto &module_symbol = *main_symbol;
```

Comment:
How do we know this unconditional dereference is safe? Is the use of a pointer the right thing at all?

Reply:
Not easy to replace by a reference, but will try.